src/Symfony/Component/Process/Process.php

```diff
@@ -1562,4 +1567,13 @@ private function escapeArgument($argument)
 
         return '"'.str_replace(array('"', '^', '%', '!', "\n"), array('""', '"^^"', '"^%"', '"^!"', '!LF!'), $argument).'"';
     }
+
+    private function replacePlaceholder($commandLine, $toReplace)
+    {
+        if (preg_match('/\{([a-zA-Z0-9]*?)\}/', $commandLine)) {
```

Comment:
Either this test is useless (because `{` or `}` chars are replaced) or the following line is buggy (because `{` and `}` are replaced regardless that they match the previous regex).

Reply:
From what I understand, it's buggy. Only placeholders like `{foo}` must be replaced. `{}` should not match, `echo '{{foo}'` should return `echo '{value'` too.

Something like this should do the trick:

```
if (!preg_match_all('/\{([a-zA-Z0-9]+?)\}/', $commandLine, $matches)) {
  return $commandLine;
}

$patterns = $replacements = array();
foreach ($matches as $match) {
  $patterns[] ='/{%s}/', $match['1']);
  $replacements = sprintf('%s%s%s', $toReplace[0], $match['1'], $toReplace['1']);
}

return preg_replace($patterns, $replacements, $commandLine);
```

(I've not tested it, but you got the idea)

Reply:
I understand and will modify according to the exemple